### ISSUE TITLE
spatial low pass filter

### ISSUE THREAD

**Issue by basaks** [GeoscienceAustralia/PyRate#156]
Matlab code for slpfilter has the following options to fill the nans.
OK to test the code with the last option?

```
% im=fillnan(ifg);
% im=inpaint_nans(double(ifg),2);
% im=fillnan_interp(ifg,'linear');
ifg(isnan(ifg))=0;
```

**Comment by mcgarth** [comment]
I think the top three options are interpolating across Nan patches. This makes more sense than converting to zeros since zeros will affect the spatial filter output?

**Comment by basaks** [comment]
ok.. let's hold onto implementing those interpolation for now. We first implement a spatial filter then if there is still time, we come back to it?

That way we will have a full implementation of the spatio-temporal filter, which can be enhanced later.

**Comment by basaks** [comment]
Spatio temporal filter status by this # edcec43ac74e07046dc0d6b2829cfd7913c32082 , we further need  

  `# need ts2ifgs equivalent`

I.e., the matab equivalent of time series to ifg conversion.

Then we are done with spatio-temporal filter, except some MPI optimisations (at the moment most functions just use single processes).

The auto cutoff from (alpha) is not matched with matlab, since we have a mismatch of `x_res/y_res` between matlab and pyrate.

It's tested in the `tests.test_aps.py` by assigning the same `x_res/y_res` as matlab in pyrate.

**Comment by basaks** [comment]
https://docs.scipy.org/doc/scipy/reference/generated/scipy.interpolate.griddata.html is what we need as a replacement for the matlab `inpaint_nans`.

**Comment by mcgarth** [comment]
Regarding the `x_res` / `y_res` discrepancy, see https://github.com/GeoscienceAustralia/PyRate/issues/24

**Comment by basaks** [comment]
Update: the `inpaint_nans`/interpolation vs zero fill functionality has been implemented using PR #167.
The ts2ifgs has been implemented already.
The auto `cutoff` frequency has been tested in pyrate vs matlab by using the same  `x_res/y_res` as in matlab.
@mcgarth close when you have merged PR #167.
